youtube_dl/extractor/common.py

```diff
@@ -2590,7 +2595,8 @@ def _media_formats(src, cur_media_type, type_info={}):
                             'url': absolute_url(src),
                         })
             for f in media_info['formats']:
-                f.setdefault('http_headers', {})['Referer'] = base_url
+                referrer = get_referrer_url(base_url, f["url"], referrer_policy)
+                f.setdefault('http_headers', {})['Referer'] = referrer
```

Comment:
the `get_referrer_url` function can return None, there is not point in setting `Referer` to None.